Tests/Tests/UserInfoProviderSpec.swift

```diff
@@ -157,6 +157,106 @@ class UserInfoProviderSpec: QuickSpec {
                     expect(userInfoProviderA == userInfoProviderB).to(beTrue())
                 }
             }
+
+            context("when checking inequality") {
+                var userInfoProviderA: UserInfoProviderMock!
+                var userInfoProviderB: UserInfoProviderMock!
+
+                beforeEach {
+                    userInfoProviderA = UserInfoProviderMock()
+                    userInfoProviderB = UserInfoProviderMock()
+                }
+
+                it("will return true for nil object") {
+                    userInfoProviderA = nil
+                    userInfoProviderB = nil
+                    expect(userInfoProviderA != userInfoProviderB).to(beFalse())
+                }
+
+                it("will return true for nil and empty objects") {
+                    userInfoProviderA = nil
+                    expect(userInfoProviderA != userInfoProviderB).to(beTrue())
+                }
+
+                it("will return false for different userIds") {
```

Comment:
But userIds are not different in this test

Reply:
Oh ya, the user info is treated by userIds in this case should return false.

Reply:
Fixed here [8b78146](https://github.com/rakutentech/ios-inappmessaging/pull/257/commits/a88a42758571ac69c74129b85cc4835ee8b78146).